pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -2541,7 +2541,25 @@ public boolean isWrapperFor(Class<?> iface) throws SQLException {
 
   public ResultSet getFunctions(String catalog, String schemaPattern, String functionNamePattern)
       throws SQLException {
-    return getProcedures(catalog, schemaPattern, functionNamePattern);
+    String sql;
+    sql = "SELECT NULL AS FUNCTION_CAT, n.nspname AS FUNCTION_SCHEM, p.proname AS FUNCTION_NAME, "
+            + "d.description AS REMARKS, "
+            + DatabaseMetaData.procedureReturnsResult + " AS FUNCTION_TYPE, "
```

Comment:
I also noticed couple of joins are not used, so I will simplified even further on the query.